### PR TITLE
PORTALS-1850: add ability to specify a primary button (text and link config)

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

**Comment by jay-hodgson** [comment]
Small enhancement to GenericCard to give ability to add CTA button (with a link configured per row).
Screen Shot 2021-02-17 at 3 19 41 PM

File: src/lib/containers/GenericCard.tsx [hunk 1]
```diff
@@ -255,21 +257,32 @@ export default class GenericCard extends React.Component<
     } else if (!titleLink) {
       target = '_blank'
     } else if (titleLink) {
+      href = this.getCardLinkHref(titleLink, data, schema) ?? ''
```
Comment (jay-hodgson) [inline src/lib/containers/GenericCard.tsx:260]: refactored this code into a shared getCardLinkHref() function, reused for the new cta button (when link is defined).

File: src/lib/containers/query_wrapper_plot_nav/QueryWrapperPlotNav.md [hunk 2]
```diff
@@ -23,6 +23,15 @@ Configured to show cards:
       //   'fundingAgency',
       //   'abbreviation',
       // ],
+    },
+    ctaButtonLinkConfig: {
```
Comment (jay-hodgson) [inline src/lib/containers/query_wrapper_plot_nav/QueryWrapperPlotNav.md:27]: Example (shown in screenshot attached to this PR)

File: src/lib/style/components/_cards.scss [hunk 3]
```diff
@@ -31,6 +31,10 @@ $text-color-lighter: #898989;
   .SRC-portalCardMain {
     display: flex;
   }
+  .SRC-portalCardCTAButton {
```
Comment (jay-hodgson) [inline src/lib/style/components/_cards.scss:34]: Ability to style this more in the future (named in the same way as other generic card elements in this scss)

**Review by emmanmills — APPROVED** [review]

**Pull request merged**